src/components/CheckboxButton/CheckboxButton.tsx

```diff
@@ -1,25 +1,38 @@
 import React from 'react';
+import Checkbox from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
 
-type CheckboxProps = {
+interface CheckboxButtonProps {
   label: string;
   checked: boolean;
-  onChange: (checked: boolean) => void;
-};
-
-export function CheckboxButton({ label, checked, onChange }: CheckboxProps) {
-  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    onChange(event.target.checked);
-  };
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  indeterminate?: boolean;
+}
 
+export const CheckboxButton: React.FC<CheckboxButtonProps> = ({
```

Comment:
I think React.FC is already deprecated, just type props the way we did for other components.